esileapclient/tests/functional/v1/test_policy.py

```diff
@@ -0,0 +1,197 @@
+                access to.
+            Test steps:
+            1) (authorized-owner) Create an offer for an owned node.
+            2) Check that offer details were returned.
+            3) (unauthorized-owner) Attempt to delete the offer from step 1.
+            4) Check that the command failed. (returned non-zero exit code)
+            5) Check that the proper error message was sent to stderr.
+            6) (authorized-owner) (cleanup) Delete the offer from step 1. """
+        offer = esi.offer_create(self.clients['authorized-owner'],
+                                 self.dummy_node.uuid,
+                                 lessee=self.projects['authorized']['name'],
+                                 resource_type='dummy_node')
+        self.assertNotEqual(offer, {})
+        self.addCleanup(esi.offer_delete,
+                        self.clients['authorized-owner'],
+                        offer['uuid'])
+
+        e = self.assertRaises(CommandFailed,
+                              esi.offer_delete,
+                              self.clients['unauthorized-owner'],
+                              offer['uuid'])
+        self.assertIn('Access was denied to offer %s.' % offer['uuid'],
+                      e.stderr.decode())
+
+    @pytest.mark.negative
+    def test_lease_delete_unauthorized(self):
+        """ Tests that an owner can't delete a lease that they do not have
+                access to.
+            Test steps:
+            1) (authorized-owner) Create a lease on an owned node.
+            2) Check that lease details were returned.
+            3) (unauthorized-owner) Attempt to delete the lease from step 1.
+            4) Check that the command failed. (returned non-zero exit code)
+            5) Check that the proper error message was sent to stderr.
+            6) (authorized-owner) (cleanup) Delete the lease from step 1. """
+        lease = esi.lease_create(self.clients['authorized-owner'],
+                                 self.dummy_node.uuid,
+                                 self.projects['authorized']['name'],
+                                 resource_type='dummy_node')
+        self.assertNotEqual(lease, {})
+        self.addCleanup(esi.lease_delete,
+                        self.clients['authorized-owner'],
+                        lease['uuid'])
+
+        e = self.assertRaises(CommandFailed,
+                              esi.lease_delete,
+                              self.clients['unauthorized-owner'],
+                              lease['uuid'])
+        self.assertIn('Access was denied to lease %s.' % lease['uuid'],
+                      e.stderr.decode())
+
+    @pytest.mark.negative
+    def test_offer_claim_unauthorized(self):
+        """ Tests that a lessee can't claim an offer that they don't have
+                access to.
+            Test steps:
+            1) (authorized-owner) Create an offer for an owned node without
+                specifying a lessee.
```

Comment:
Do you mean "that does specify a lessee"?

Reply:
Yep, I think I wrote that spec before I knew the intended behavior for offer creation lol. Will fix!